### ISSUE TITLE
Portals [React-Compat] missing field and rendering problem

### ISSUE THREAD

**Issue by wahur666** [preactjs/preact#3507]
[+] Check if updating to the latest Preact version resolves the issue

React Portals has a property called "containerInfo" which can return the container itself. This is documented somewhere for sure, but I only found out from the Dart docs. https://pub.dev/documentation/react/latest/react_client.react_interop/ReactPortal-class.html
This problem occurred while integrating with this proprietary library: 

https://bryntum.com/docs/scheduler/#  
https://bryntum.com/docs/scheduler/guide/Scheduler/integration/react/guide#rendering-react-components-in-column-cells

Where there is a React wrapper for the components, which uses Portals to move cells around. Relevant snippet:

```js 
if (portal) {
    // Move portal container back to the cell if we have one
    renderElement.appendChild(portal.containerInfo);
    renderElement.portalContainer = portal.containerInfo;
} else {
    // Create new portal container
    ...
    renderElement.portalContainer = _portalContainer; // Create a new portal in the portal container
    portal = ReactDOM.createPortal(reactNode, _portalContainer, portalId); // Add the new portal to Map
```

Here is what React produces:

![kép](https://user-images.githubusercontent.com/9078042/162213665-0bd96e12-a765-4a1d-9e74-cd785a8046ea.png)

A portal, with a component inside, with its children.

What Preact produces:

![kép](https://user-images.githubusercontent.com/9078042/162214018-d9636801-df01-41b5-9ad7-57776afb8ceb.png)

The container is okay, it but the children are missing from it:

![kép](https://user-images.githubusercontent.com/9078042/162214542-913cfa08-aac2-4c84-a9bc-492093fff551.png)

Which is also part of the component:

![kép](https://user-images.githubusercontent.com/9078042/162214675-9875c5ce-e520-4590-97b2-62cbe356b65e.png)

The props children never shows up on the output.

**Expected behavior**

The compat library to incorporate this functions.

** Steps to reproduces ** 

1. Get the 45 day trial, and based of the doc, just override the cell renderer method with JSX

** Partial solution **

During debugging I already managed to link up the missing field with this code snippet, but this is sub optimal.

compat/src/portals.js
```js
export function createPortal(vnode, container) {
	return createElement(Portal, { _vnode: vnode, _container: container, _expose: "containerInfo" });
}
```

compat/src/render.js
```js
let oldVNodeHook = options.vnode;
options.vnode = vnode => {
	let type = vnode.type;
	let props = vnode.props;
	let normalizedProps = props;
	if (vnode.props._expose === "containerInfo") {
		vnode.containerInfo = vnode.props._container;
	}
```

I can do more debugging, just the output of the preact bundle is minified, and thats not ideal tracking whats not working as expected.

Any help would be useful, and would prevent a React switch. :)

**Comment by wahur666** [comment]
Local recreation: 
compat/test/browser/portals.test.js
```js
	it('should render into a different root node', () => {
		let root = document.createElement('div');
		document.body.appendChild(root);

		const A = () => <span>A</span>

		function Foo(props) {
			const portal = createPortal(props.children, root);
			return <div>{portal}</div>;
		}
		render(<div><A /></div>, scratch);

		expect(root.innerHTML).to.equal('<span>A</span>');

		root.parentNode.removeChild(root);
	});
```

```
expected '' to equal '<span>A</span>'
Expected :"<span>A</span>"
Actual   :""
```

**Comment by JoviDeCroock** [comment]
@wahur666 your test has a typo, I think you meant to do this

```js
	it('should render into a different root node', () => {
		let root = document.createElement('div');
		document.body.appendChild(root);

		const A = () => <span>A</span>

		function Foo(props) {
			const portal = createPortal(props.children, root);
			return <div>{portal}</div>;
		}
		render(<Foo><A /></Foo>, scratch);

		expect(root.innerHTML).to.equal('<span>A</span>');

		root.parentNode.removeChild(root);
	});
```

As you are not using `Foo` so we can't render anything, this test succeeds

**Comment by wahur666** [comment]
> @wahur666 your test has a typo, I think you meant to do this
> 
> ```js
> 	it('should render into a different root node', () => {
> 		let root = document.createElement('div');
> 		document.body.appendChild(root);
> 
> 		const A = () => <span>A</span>
> 
> 		function Foo(props) {
> 			const portal = createPortal(props.children, root);
> 			return <div>{portal}</div>;
> 		}
> 		render(<Foo><A /></Foo>, scratch);
> 
> 		expect(root.innerHTML).to.equal('<span>A</span>');
> 
> 		root.parentNode.removeChild(root);
> 	});
> ```
> 
> As you are not using `Foo` so we can't render anything, this test succeeds

Yes, my mistake, so I cannot create it on locally still. :/

**Comment by wahur666** [comment]
Here is more info, that I can extract, since I cannot share the code itself.
![kép](https://user-images.githubusercontent.com/9078042/162224019-3664d69a-a2c2-44a7-841b-13b129c87570.png)
![kép](https://user-images.githubusercontent.com/9078042/162224054-c6fcf8da-6a00-4763-b473-102da1c9fd2a.png)
![kép](https://user-images.githubusercontent.com/9078042/162224176-dd14a71c-97ac-4975-90b0-36755a902bae.png)

**Comment by wahur666** [comment]
Sorry, here is my problem visually: 
![kép](https://user-images.githubusercontent.com/9078042/162228760-03ac198a-67ca-4534-b242-eea769b9b10c.png)
Preact on the left side, does not render the same element what React does.

**Comment by JoviDeCroock** [comment]
The screenshots aren't really telling me why that's happening but given your initial code-snippet I assume it's because of missing the `containerInfo` property? Would this addition fix it? like https://github.com/preactjs/preact/pull/3508

**Comment by wahur666** [comment]
> The screenshots aren't really telling me why that's happening but given your initial code-snippet I assume it's because of missing the `containerInfo` property? Would this addition fix it? like #3508

Let me start again. Now with a bit more clarity hopefully. This 3rd party proprietary library using portals to render the cells one by one. Where you can pass a custom react function to customize it. Now the problem is besides the missing field, what I also patched on my version. That when preact does the portal rendering, the custom react function for customization does not rendered into the portal container itself. 

![kép](https://user-images.githubusercontent.com/9078042/162230927-3bdf5449-8ded-4747-933e-cefc3d2af74c.png)

The function what should appear on both sides are the following:

```js
({ record }) =><div>{JSON.stringify(record)}</div>
``` 

The code snippets what I shared are part of the 3rd party library, and I don't know, how much in my liberty to share it. 
But the behavior what I examined so far, that the plain HTML element container does not get the preact rendered output. I'm currently trying to make a stripped down example code, but still cannot reproduce it.
